Awful.apk/src/main/java/com/ferg/awfulapp/ThreadDisplayFragment.java

```diff
@@ -779,10 +781,28 @@ public void failure(VolleyError error) {
 		  .setNegativeButton(R.string.cancel, null)
 		  .show();
 	}
-	
-    @Override
-    public void onSaveInstanceState(@NonNull Bundle outState){
-    	super.onSaveInstanceState(outState);
+
+	/**
+	 * Toggles whether to display this user's avatar or not
+	 *
+	 * @param avatarUrl		The URL of the avatar to toggle the display of. Ignored if null or empty string.
+	 */
+	public void toggleAvatar(String avatarUrl) {
+		if (TextUtils.isEmpty(avatarUrl)) {
+			return;
+		}
+		Set<String> blocked = getPrefs().getPreference(Keys.BLOCKED_AVATAR_URLS, Collections.emptySet());
+		Set<String> newSet = new HashSet<>(blocked); // not allowed to mutate original set
+
+		if (!newSet.remove(avatarUrl)) {
+			newSet.add(avatarUrl);
```

Comment:
It seems kinda useful to be able to block an exact avatar, in case it's a redtext av or something. Would it be better to change the language in the menu, like "Block exact avatar" and "Show [blank]'s avatar", or would changing to the user ID be the preferred way to go?